packages/grid/_modules_/grid/models/gridExport.ts

```diff
@@ -1,7 +1,28 @@
+/**
+ * The options to apply an export.
+ */
+export interface GridExportOptions {
+  /**
+   * The columns exported.
+   * This should only be used if you want to restrict the columns exports.
+   */
+  fields?: string[];
+  /**
+   * If `true`, the hidden columns will also be exported.
+   * @default false
+   */
+  allColumns?: boolean;
+  /**
+   * If `true`, this export option will be removed from the GridToolbarExport menu.
```

Comment:
What should be the behavior if the user calls `exportDataAsCsv` with `disable: true` ? (same for print)
Should be raise an error ? Ignore it as you currently do ? Skip the export ?

The most coherent behavior would probably be to split the interfaces between the imperative method and the React component, to only have `disable` on  the React component.

This would be an argument in favor of `csvOptions={false}` instead of `csvOptions={{ disable: false }}`

Reply:
Yes, if `GridCsvExportOptions` is used by imperative methods, it does not make sens to put user interface customisation in it.

`disable` could be defined in type `GridExportOption`. But it does not make sens to add an option `disable` that will disable all the other options.

Does someone else has an argument in favour of the `disable` logic?

Reply:
I'm not very in favor of `csvOptions={false}`, since the prop was meant to be an object initially. We have been using slots to allow simple customizations without forcing to override an entire component. For instance, the options of the column menu can be easily changed doing that: https://mui.com/components/data-grid/components/#column-menu. The same approach could be done here maybe. It would even allow users to add their export options.

Reply:
My idea is not exactly as you proposed. We add a new `toolbarExportButton` slot which has by default:

```tsx
<GridToolbarExportContainer>
  <GridCsvExportItem options={...} />
  <GridPrintExportItem options={...} />
  <MyCustomExportItem options={...} />
</GridToolbarExportContainer>
```

If the user wants to disable one of item, he removes the respective option. My argument against passing `false` is that the options prop is meant to be object. We're changing its meaning depending on its value. If the slot approach is too overkill, we could also add an `allowedFormats` prop to the `GridToolbarExportContainer`.

Reply:
Or `disableToolbarButton`